arch/x86/core/intel64/cpu.c

```diff
@@ -121,6 +122,16 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	uint8_t vector = ((unsigned long) x86_ap_start) >> 12;
 	uint8_t apic_id = x86_cpu_loapics[cpu_num];
 
+	if (IS_ENABLED(CONFIG_ACPI)) {
+		struct acpi_cpu *cpu;
+
+		cpu = z_acpi_get_cpu(cpu_num);
+		if (cpu != NULL) {
+			/* ToDo: check the flags */
```

Comment:
what are the implications of not checking them? suggest filing a GH enhancement if this is something we really want to circle back on.

Reply:
Actually I fixed this by checking the flags in z_acpi_get_cpu(). In my up², I end up with 4 CPUs in the MADT, with 2 disabled.

Reply:
Btw, the documentation is not totally clear about the hierarchy of CPU (BSP vs APs). It says ""The MP initialization protocol defines two classes of processors: the bootstrap processor (BSP) and the application processors (APs). Following a power-up or RESET of an MP system, system hardware dynamically selects one of the processors on the system bus as the BSP. The remaining processors are designated as APs."

Does that mean the BSP could be any of the 2 cores then? So not specifically the first CPU in the ACPI MADT?
I quickly checked on my up², getting the bsp apic_id via cpuid instruction. Ok it matches the order we hard-coded in zephyr, but could it be that such order may not be this one? (Or is are the apic_id set after the BSP/APs mitigation process was done... could not find a clear answer on that one).